homeassistant/components/poolsense/coordinator.py

```diff
@@ -3,26 +3,34 @@
 import asyncio
 from datetime import timedelta
 import logging
+from typing import TYPE_CHECKING
 
 from poolsense import PoolSense
 from poolsense.exceptions import PoolSenseError
 
+from homeassistant.const import CONF_EMAIL
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.typing import StateType
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import DOMAIN
 
+if TYPE_CHECKING:
+    from . import PoolSenseConfigEntry
```

Comment:
This wont work, CI is also complaining. You need to define the `PoolSenseConfigEntry` here and import it in `__init__.py` to avoid circular imports like e.g. Shelly

Reply:
Hmm, this should work, more integrations have this approach:

https://github.com/home-assistant/core/blob/6cb703aa1d556fdd342df18db23637062c2547c3/homeassistant/components/trafikverket_weatherstation/coordinator.py#L24-L34

Reply:
It is probably because `from __future__ import annotations` is not included so you either need to import that or do `config_entry: "TVWeatherConfigEntry"`

Reply:
Oh that first thing is probably it